### PR TITLE
Fixing identity client request content type

### PR DESCRIPTION
**Pull request by schaabs**

### CODE REVIEW

File: sdk/keyvault/Azure.Security.KeyVault.Secrets/tests/SecretsTests.cs [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+using System.Linq;
+using Azure.Security.KeyVault.Secrets;
+using NUnit.Framework;
+using Azure.Identity;
+
+namespace Azure.Security.KeyVault.Test
+{
+    public class SecretTests : KeyVaultTestBase
+    {
+        //private static string s_tenantId = "72f988bf-86f1-41af-91ab-2d7cd011db47";
```
Comment (AlexGhiondea) [inline sdk/keyvault/Azure.Security.KeyVault.Secrets/tests/SecretsTests.cs]: We should remove these.

File: sdk/keyvault/Azure.Security.KeyVault.Secrets/tests/SecretsTests.cs [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+
+            Secret setResult = await client.SetAsync("CrudBasic", "CrudBasicValue1");
+
+            Secret getResult = await client.GetAsync("CrudBasic");
+
+            AssertSecretsEqual(setResult, getResult);
+
+            DeletedSecret deleteResult = await client.DeleteAsync("CrudBasic");
+
+            //HACK: AssertSecretsEqual(setResult, deleteResult);
```
Comment (AlexGhiondea) [inline sdk/keyvault/Azure.Security.KeyVault.Secrets/tests/SecretsTests.cs]: This should be uncommented

File: sdk/identity/Azure.Identity/src/IdentityClient.cs [hunk 3]
```diff
@@ -69,20 +70,19 @@ private Request CreateClientSecretAuthRequest(string tenantId, string clientId,
...
+
+            request.Headers.SetValue("Content-Type", "application/x-www-form-urlencoded");
 
             request.UriBuilder.Uri = _options.AuthorityHost;
 
             request.UriBuilder.AppendPath(tenantId);
 
             request.UriBuilder.AppendPath("/oauth2/v2.0/token");
 
-            ReadOnlyMemory<byte> content = Serialize(
```
Comment (pakrym) [inline sdk/identity/Azure.Identity/src/IdentityClient.cs]: Do we still need the Serialize method?

Comment (pakrym) [reply sdk/identity/Azure.Identity/src/IdentityClient.cs]: FixedSizedBufferWriter can go too

Comment (schaabs) [reply sdk/identity/Azure.Identity/src/IdentityClient.cs]: removed

**Review by AlexGhiondea — APPROVED** [review]

**Pull request merged**